src/service.rs

```diff
@@ -0,0 +1,254 @@
+use crate::auth::method::{AuthResult, AuthMethod};
+use crate::config::VaultConfig;
+
+use async_trait::async_trait;
+use futures::executor::block_on;
+use serde::{Deserialize, Serialize};
+use std::{error::{self, Error}, sync::Arc};
+use tokio::{sync::{RwLock, mpsc::{Receiver, Sender, channel}}, task::JoinHandle};
+use vaultrs::{
+    client::{Client, VaultClient, VaultClientSettingsBuilder},
+    kv2,
+    api::kv2::responses::SecretVersionMetadata
+};
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct HealthCheckData {
+    pub(crate) data: String,
+}
+
+pub struct VaultService {
+    pub(crate) client: Arc<RwLock<VaultClient>>,
+    pub(crate) auth_result: AuthResult,
+    pub(crate) mount_path: String,
+    pub(crate) healthcheck_file_path: String,
+}
+
+impl std::fmt::Debug for VaultService {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("VaultService")
+        .field("auth_result", &self.auth_result)
+        .finish()
+    }
+}
+
+impl VaultService {
+    pub async fn default() -> Result<(Self, Arc<RwLock<dyn AuthMethod>>), Box<dyn error::Error>> {
```

Comment:
I think we should rename this to `fromEnv` to make it clear it's loading values from the environment.